### ISSUE TITLE
Shulker Boxes clearing contents

### ISSUE THREAD

**Issue by der1cho** [ricksouth/serilum-mc-mods#2800]
Minecraft version: 1.21.4
Modloader: Fabric
Fabric loader version: 0.16.9
Environment: Singleplayer

I updated the pack to 1.21.4 for The Pale Forest update and after traveling some 40k blocks to find a pale forest I filled up a Shulker Box with items, took to box back to my spawn and the items had cleared. Thought I either had left the box behind or maybe an issue with the distance travelled, returned to destination and opened my firework shulker to get some rockets and they were also cleared. Have also stored pale forest items in an Ender Chest and returned to spawn but once I put them into a Shulker, they vanished. Also have lost grass and natural contents from another shulker which I use for landscaping and a near full shulker box of firework rockets I use for Elytra travel.

So in short, lost contents from what was an empty shulker as well as two long established shulkers.

**Comment by James103** [comment]
Which mods are you referring to? Can you please post the full mod list or attach a `latest.log` from reproducing this issue?

**Comment by der1cho** [comment]
I'm using The Vanilla Experience modpack via Curseforge, haven't modified anything within the pack and have had no issues in the past versions.

**Comment by der1cho** [comment]
[latest.log](https://github.com/user-attachments/files/18139024/latest.log)

If that works, will be as said the latest log.

Went in, gathered material to fill a shulker box of rockets from my storage system and they immediately vanished after filling and breaking the box to get in my inventory.

**Comment by James103** [comment]
It sounds like breaking a shulker box doesn't properly copy the contents of the shulker box block entity to the shulker box item, causing the shulker box's contents to be voided. Can you please test this by placing down a shulker box, putting a piece of cobblestone, dirt, or any other basic item inside, then breaking the shulker box by hand (with a pickaxe)?

Similarly, does a piston breaking a shulker box void the contents of that shulker box?

**Comment by der1cho** [comment]
Done some tests (pickaxe and piston breaking works okay) and have come across another issue and a potential quirk which triggers the lost of items.

There is an option to open the Shulker without placing it down, that has triggered the loss of items.

The issue is I don't have the option to make a fresh Shulker box. I tried to make one to test the option of using a newly made Shulker but can't make one.

**Comment by der1cho** [comment]
Is there an update on this situation, even if it's only in regard with the inability to craft a new Shulker Box?

**Comment by ricksouth** [comment]
I'm really sorry for the inconvenience. This was a bug for 1.21.4 specifically. I have just uploaded version 1.7 of Quick Right-Click that fixes the issue.

https://www.curseforge.com/minecraft/mc-mods/quick-right-click/files
